File: assign03/primes.pl
# This is my first Perl program. 
sub isPrime {
	$n = scalar(@_[0]);
	@primeArr = ();
	for ($i = 0; $i <= $n; $i++) {
		push(@primeArr, 1);
	}
	for ($i = 2; $i * $i <= $n; $i++) {
		if (@primeArr[$i] == 1) {
			for ($k = $i * 2; $k <= $n; $k = $k + $i) {
				@primeArr[$k] = 0;
			}
		}
	}
	return @primeArr[$n];
}

for ($l = 2; $l <= 100; $l++) {
	if (isPrime($l) == 1) {
		print $l, " ";
	}
}
print "\n";

File: assign03/primes.php
<?php
/* This is my first PHP program. */

function isPrime($x){
	$primeArr = array();
	for ($i = 2; $i <= $x; $i++) {
		$primeArr[$i] = true;
	}
	for ($i = 2; $i * $i <= $x; $i++){
		if ($primeArr[$i] == true) {
			for ($k = $i * 2; $k <= $x; $k += $i){
				if ($k == $x) {
					return false;
				}
				$primeArr[$k] = false;
			}
		}
	}
	
	return $primeArr[$x];
}

for ($i = 2; $i <= 100; $i++) {
	if (isPrime($i)) {
		echo $i, " ";
	}
}
echo "\n";

?>
